com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -978,13 +1010,16 @@ private void downloadSymbols(IProgress progress) throws IOException, CompileExce
                 }
                 case "clean": {
                     IProgress m = monitor.subProgress(1);
-                    m.beginTask("Cleaning...", newTasks.size());
-                    for (Task<?> t : newTasks) {
-                        List<IResource> outputs = t.getOutputs();
-                        for (IResource r : outputs) {
-                            BundleHelper.throwIfCanceled(monitor);
-                            r.remove();
+                    List<String> paths = state.getPaths();
+                    m.beginTask("Cleaning...", paths.size());
+                    for (String path : paths) {
+                        File f = new File(path);
+                        if (f.exists()) {
+                            state.removeSignature(path);
+                            f.delete();
                             m.worked(1);
+                            System.out.printf("Deleted %s", path);
+                            BundleHelper.throwIfCanceled(monitor);
```

Comment:
Previously, we created a full tree of the tasks (thus reading them!) to figure out the complete tree.

But, we also already had the "state" file with path:signature mappings, so I instead use that now, and we no longer have the need to build (!) before we clean content.

Reply:
Nice!